### ISSUE TITLE
Audit async/await

### ISSUE THREAD

**Issue by crutkas** [Microsoft/WindowsTemplateStudio#851]
After #844, i think we should do a full audit.

**Comment by crutkas** [comment]
this is what i found inside the Big.sln, few async voids too


code\src\UI\VisualStudio\SolutionWizard.cs -> RunFinished
RunFinished also is async void
SolutionWizard has a lot of non-referenced methods



code\src\UI\ViewModels\NewProject\MainViewModel.cs(166)
async void OnTemplatesAvailable()
async void OnNewTemplatesAvailable()
Both are Async void

code\src\UI\ViewModels\Common\BaseMainViewModel.cs(283)
async void OnCheckUpdates()

code\src\Installer.2017\Commands\RelayCommandPackage.cs(66)
async Task<IGenContextBootstrapService> PrepareBootstrapSvc()
async Task<object> CreateService(IAsyncServiceContainer container, CancellationToken cancellationToken, Type serviceType)


code\src\Installer.2017\Commands\GenContextBootstrapService.cs(42)
public async System.Threading.Tasks.Task GenContextInit()

code\test\Templates.Test\StyleCopProjectGenerationTests.cs(69)
async void GenerateAllPagesAndFeaturesAndCheckWithStyleCop(string projectType, string framework)

code\test\Templates.Test\ProjectGenerationTests.cs(150)
async void GenerateAllPagesAndFeaturesRandomNames(string projectType, string framework)
async void GenerateAllPagesAndFeatures(string projectType, string framework)
async void GenerateProjectWithIsolatedItems(string itemName, string projectType, string framework, string itemId)
async void GenerateEmptyProject(string projectType, string framework)
async void GenerateEmptyProject(string projectType, string framework)


code\src\Core\Locations\TemplatesSynchronization.cs(206)
async Task CheckContentUnderVersion()
async Task CheckContentOverVersion()
async Task CheckInstallDeployedContent()
async Task Do()

code\src\Core\Diagnostics\TelemetryTracker.cs(146)
async Task TrackEditSummaryItem(EditItemActionEnum trackedAction)

code\test\VsEmulator\Main\MainViewModel.cs(179)
async void NewProject()

test\Core.Test\TestData\Merge\Source_expected.cs(93):
async void OnLaunched(LaunchActivatedEventArgs e)

code\test\Core.Test\TestData\Merge\Source.cs(88)
async void OnLaunched(LaunchActivatedEventArgs e)

**Comment by onovotny** [comment]
You don't need to do this by hand.... Use this: https://www.nuget.org/packages/Microsoft.VisualStudio.Threading.Analyzers

Calls these out in the IDE. You can also configure them as either warnings or errors if you want to hard-fail.

**Comment by crutkas** [comment]
Zomg. I need to chat with you more often!

**Comment by crutkas** [comment]
@mrlacey how does the analyzers work for the stylecop checks.  we should add this in as well there.

**Comment by mrlacey** [comment]
@crutkas Can add that package to the Secret StyleCop template, set warnings as errors and it would be caught as part of the test....i think

**Comment by mrlacey** [comment]
I've added the analyzer to the main projects and fixed the naming issues and a few easy other things. It's also highlighted a bunch of other issues that are more involved and will take longer to fix than I have time in the next week so leaving this open.
Need to address or suppress (with good reasons and comments) the warnings raised by the analyzer.

I tried adding the analyzer to the StyleCop tests but as the Analyzer covers a lot of other things it found a lot of false positives that show this isn't the best way to verify the naming in the generated projects. Have created #969 to track that as a separate issue.

**Comment by crutkas** [comment]
with #1002, this can be closed out, no?

**Comment by crutkas** [comment]
ahem, @mrlacey with #1002, this can be closed?

**Comment by mrlacey** [comment]
This is still raising some warnings, but they're complicated to fix. 

>  It's also highlighted a bunch of other issues that are more involved and will take longer to fix than I have time in the next week so leaving this open.
Need to address or suppress (with good reasons and comments) the warnings raised by the analyzer.

Could close this and raise another issue to follow up but had left this open and unassigned to address this here. Could close on the basis that the audit has been done but this feels just like moving the work along.

**Comment by mrlacey** [comment]
don't see a reason to keep this open
